charts/generic-service/templates/deployment.yaml

```diff
@@ -179,6 +179,15 @@ spec:
               value: {{ .Values.tracing.probability | quote }}
             - name: JAEGER_PROPAGATION
               value: b3  # Use B3 propagation for Istio compatibility
+            # OpenTelemetry support
+            - name: OTEL_SERVICE_NAME
+              value: '{{ include "generic-service.name" . }}.{{ .Release.Namespace }}'  # Match naming scheme of Istio
+            - name: OTEL_TRACES_SAMPLER
+              value: parentbased_traceidratio
+            - name: OTEL_TRACES_SAMPLER_ARG
+              value: {{ .Values.tracing.probability | quote }}
+            - name: OTEL_PROPAGATORS
+              value: b3  # Use B3 propagation for Istio compatibility
```

Comment:
Istio uses `x-b3-*` headers ([docs](https://istio.io/latest/docs/tasks/observability/distributed-tracing/overview/)), so this should probably be `b3multi` instead of `b3`.

Note: `JAEGER_PROPAGATION` should stay `b3` because `b3multi` is not a valid value there ([docs](https://www.jaegertracing.io/docs/1.24/client-features/)).